src/contrib/persistence/Akka.Persistence.Sql.Linq2Db/Journal/Linq2DbWriteJournal.cs

```diff
@@ -0,0 +1,189 @@
+                    throw;
+                }
+
+                if (_journalConfig.TableConfig.AutoInitialize)
+                {
+                    try
+                    {
+                        _journal.InitializeTables();
+                    }
+                    catch (Exception e)
+                    {
+                        Context.GetLogger().Warning(e,
+                            "Unable to Initialize Persistence Journal Table!");
+                    }
+
+                }
+            }
+            catch (Exception ex)
+            {
+                Context.GetLogger().Warning(ex,"Unexpected error initializing journal!");
+            }
+        }
+
+        protected override bool ReceivePluginInternal(object message)
+        {
+            if (message is WriteFinished wf)
+            {
+                writeInProgress.Remove(wf.PersistenceId);
+            }
+            else
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public override void AroundPreRestart(Exception cause, object message)
+        {
+            Context.System.Log.Error(cause,
+                $"Linq2Db Journal Error on {message?.GetType().ToString() ?? "null"}");
+            base.AroundPreRestart(cause, message);
+        }
+
+        
+        public override async Task ReplayMessagesAsync(IActorContext context, string persistenceId,
+            long fromSequenceNr, long toSequenceNr, long max, Action<IPersistentRepresentation> recoveryCallback)
+        {
+            await _journal.MessagesWithBatch(persistenceId, fromSequenceNr,
+                    toSequenceNr, _journalConfig.DaoConfig.ReplayBatchSize,
+                    Option<(TimeSpan, IScheduler)>.None)
+                .Take(max).SelectAsync(1,
+                    t => t.IsSuccess
+                        ? Task.FromResult(t.Success.Value)
+                        : Task.FromException<ReplayCompletion>(
+                            t.Failure.Value))
+                .RunForeach(r =>
+                {
+                    recoveryCallback(r.repr);
+                }, _mat);
+            
+        }
+
+        public override async Task<long> ReadHighestSequenceNrAsync(string persistenceId, long fromSequenceNr)
+        {
+            if (writeInProgress.ContainsKey(persistenceId))
+            {
+                try
+                {
+                    await writeInProgress[persistenceId];
+                }
+                catch (Exception)
+                {
+                    //We don't have 'Recover' in C# so this is intentionally empty
+                    //Basically we just wanted to wait for write to succeed OR fail
+                }
+                var hsn =await _journal.HighestSequenceNr(persistenceId,
+                    fromSequenceNr);
+                return hsn;
+            }
+            return await _journal.HighestSequenceNr(persistenceId, fromSequenceNr);
```

Comment:
Honestly, not sure if this was the right/best way to port intent, original [here](https://github.com/akka/akka-persistence-jdbc/blob/master/core/src/main/scala/akka/persistence/jdbc/journal/JdbcAsyncWriteJournal.scala#L104-L113)